src/main/scala/org/zalando/spearheads/innkeeper/utils/HttpClient.scala

```diff
@@ -0,0 +1,67 @@
+package org.zalando.spearheads.innkeeper.utils
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.headers.{ Authorization, OAuth2BearerToken }
+import akka.http.scaladsl.model.{ HttpHeader, HttpMethod, HttpMethods, HttpRequest }
+import akka.stream.ActorMaterializer
+import com.google.inject.Inject
+import org.slf4j.LoggerFactory
+import spray.json._
+
+import scala.collection.immutable.Seq
+import scala.concurrent.{ Await, ExecutionContext }
+import scala.concurrent.duration._
+import scala.util.Try
+
+trait HttpClient {
+
+  def callJson(uri: String,
+               token: Option[String] = None,
+               method: HttpMethod = HttpMethods.GET): Try[JsValue]
+}
+
+class AkkaHttpClient @Inject() (implicit val actorSystem: ActorSystem,
+                                implicit val materializer: ActorMaterializer,
+                                implicit val executionContext: ExecutionContext) extends HttpClient {
+
+  import scala.util.Try
+
+  val logger = LoggerFactory.getLogger(this.getClass)
+
+  override def callJson(uri: String,
+                        token: Option[String] = None,
+                        method: HttpMethod = HttpMethods.GET): Try[JsValue] = {
+
+    val futureResponse = Http().singleRequest(
+      HttpRequest(
+        uri = uri,
+        method = method,
+        headers = headersForToken(token)
+      )
+    )
+
+    val futureJsonString = futureResponse.map { res =>
+      res.entity.dataBytes.map(bs => bs.utf8String).runFold("")(_ + _)
+    }.flatMap(identity(_))
+
+    for {
+      jsonString <- Try {
+        logger.debug(s"We call the OAuth Token Info Service with the url: ${uri}")
+        Await.result(futureJsonString, 1.second)
```

Comment:
From a non-blocking/akka perspective waiting here and actually blocking is not that nice. As a todo for later, it would be nicer, to stay inside the Future instead of blocking waiting for the result.